src/chips/utils.rs

```diff
@@ -8,120 +10,90 @@ use num_bigint::BigUint;
 /// * DEG_DVD is the degree of the dividend
 /// * DEG_DVS is the degree of the divisor
 /// * Q is the modulus of the Ring. All the coefficients will be in the range [0, Q-1]
-/// * Assumes that coefficients of the dividend and divisor are u64 values
 pub fn div_euclid<const DEG_DVD: usize, const DEG_DVS: usize, const Q: u64>(
-    dividend: &Vec<u64>,
-    divisor: &Vec<u64>,
-) -> (Vec<u64>, Vec<u64>) {
-    if divisor.is_empty() || divisor.iter().all(|&x| x == 0) {
+    dividend: &Vec<BigInt>,
```

Comment:
Is there any reason why you change the type from `u64` to `BigInt`? From my understanding coeffcients are less than the field Q, which is less than `2^64-1`, hence I thought u64 would be sufficient to represent these values.